plugin/core/workspace.py

```diff
@@ -63,7 +63,7 @@ def __init__(self, window: sublime.Window) -> None:
         self._update_exclude_patterns(self.folders)
 
     def _update_exclude_patterns(self, folders: List[str]) -> None:
-        self._folders_exclude_patterns = []
+        self._folders_exclude_patterns = [[]] * len(folders)
         project_data = self._window.project_data()
```

Comment:
I would definitely put a comment on that line, or link to the issue,
it was not obvious why it needs to be multiplied by the length of the folders

Reply:
`[x] * n` is a concise syntax in Python to create a list with n entries which are all x, i.e. here it will be for example `[[], [], []]` with 3 folders.
But yeah, I was also surprised that this bug can happen, and I never saw it with "normal" usage. The `window.project_data()` even returns a dict with the (sidebar) folders, if there is no project at all (iirc ST automatically creates a "hidden" project in that case), or with a single .sublime-workspace file. It might be possible with multiple workspace files or if you edit them manually.
But with this initialization line we are certain to be on the safe side, so it looks fine to me.

Reply:
Note that those lists created via `[[]] * len(folders)` refer to exactly the same thing. Not sure whether this is what you want but just provide a potential issue.

```
>>> a = [[]] * 4
>>> a[0].append('foo')
>>> a
[['foo'], ['foo'], ['foo'], ['foo']]
```

Reply:
It should be fine, because each entry gets replaced completely, not appended to:
```python
self._folders_exclude_patterns[i] = exclude_patterns
```

For example
```
>>> a = [[]] * 4
>>> a[1] = ['foo']
>>> a
[[], ['foo'], [], []]
```